REST API: Clusters Queues list/get documentation (#2155)
buildkite/docs: pages/apis/rest_api/clusters.md

    "color": "#FFE0F1",
    "url": "http://api.buildkite.com/v2/organizations/acme-inc/clusters/42f1a7da-812d-4430-93d8-1cc7c33a6bcf",
    "web_url": "http://buildkite.com/organizations/acme-inc/clusters/42f1a7da-812d-4430-93d8-1cc7c33a6bcf",

>     "queues_url": "http://buildkite.com/organizations/acme-inc/clusters/42f1a7da-812d-4430-93d8-1cc7c33a6bcf/queues",

Reviewer: ❤️

  <tr><th><code>422 Unprocessable Entity</code></th><td><code>{ "message": "Reason the cluster couldn't be deleted" }</code></td></tr>
</tbody>
</table>

## Cluster queues

[Cluster queues](https://buildkite.com/docs/agent/clusters#set-up-a-cluster-set-up-queues) are discrete groups of agents within a cluster. Pipelines in that cluster can target cluster queues to run jobs on agents assigned to those queues.

### Cluster queue data model

<table class="responsive-table">
  <tbody>
    <tr><th><code>id</code></th><td>ID of the queue</td></tr>
    <tr><th><code>graphql_id</code></th><td><a href="/docs/apis/graphql-api#graphql-ids">GraphQL ID</a> of the queue</td></tr>
    <tr><th><code>key</code></th><td>The queue key</td></tr>
    <tr><th><code>description</code></th><td>Description of the queue</td></tr>
    <tr><th><code>url</code></th><td>Canonical API URL of the queue</td></tr>
    <tr><th><code>web_url</code></th><td>URL of the queue on Buildkite</td></tr>
    <tr><th><code>cluster_url</code></th><td>API URL of the cluster the queue belongs to</td></tr>

>     <tr><th><code>dispatch_paused</code></th><td>Indicates whether the queue is paused</td></tr>

Reviewer 2: Since "dispatching" jobs is a bit of a new concept coming in with these changes, I think it would be good to use that word here. Maybe:

```suggestion
    <tr><th><code>dispatch_paused</code></th><td>Indicates whether the queue is paused from dispatching jobs to agents</td></tr>
```

What do you think?

Author: @mbelton-buildkite something about "paused from" that I'm not sure about... what do you think of this: 

"Indicates whether job dispatch to associated agents is paused for this queue"

Reviewer 2: That could work. The only hesitations I have with it are "job dispatch" being a thing rather than an action and moving "queue" so far back in the description. I think it makes it sound a bit more complicated than it is. Another option could be:

```suggestion
    <tr><th><code>dispatch_paused</code></th><td>Indicates whether the queue has paused dispatching jobs to agents</td></tr>
```

I'm happy for you to decide which one to use. 😊

Author: Thanks @mbelton-buildkite ! How about:
"Indicates whether the queue has paused dispatching jobs to associated agents" ? Im keen to put 'associated' in there so it doesnt read as though _all_ agents are affected

    <tr><th><code>dispatch_paused_by</code></th><td>User who paused the queue</td></tr>
    <tr><th><code>dispatch_paused_at</code></th><td>When the queue was paused</td></tr>
    <tr><th><code>dispatch_paused_note</code></th><td>The note left when the queue was paused</td></tr>
    <tr><th><code>created_at</code></th><td>When the queue was created</td></tr>
    <tr><th><code>created_by</code></th><td>User who created the queue</td></tr>
  </tbody>
</table>

### List queues

Returns a [paginated list](<%= paginated_resource_docs_url %>) of a cluster's queues.

```bash
curl "https://api.buildkite.com/v2/organizations/{org.slug}/clusters/{cluster.id}/queues"
```

```json
[{
  "id": "01885682-55a7-44f5-84f3-0402fb452e66",
  "graphql_id": "Q2x1c3Rlci0tLTQyZjFhN2RhLTgxMmQtNDQzMC05M2Q4LTFjYzdjMzNhNmJjZg==",
  "key": "Default Queue",
  "description": "The default queue for this cluster",
  "url": "http://api.buildkite.com/v2/organizations/test/clusters/42f1a7da-812d-4430-93d8-1cc7c33a6bcf/queues/01885682-55a7-44f5-84f3-0402fb452e66",
  "web_url": "http://buildkite.com/organizations/test/clusters/42f1a7da-812d-4430-93d8-1cc7c33a6bcf/queues/01885682-55a7-44f5-84f3-0402fb452e66",

>   "cluster_url": "http://api.buildkite.com/v2/organizations/test/clusters/#{cluster.id}",

Reviewer 2: Should this URL contain an example cluster ID instead of `#{cluster.id}`?

Reviewer 2: If yes, it's repeated in the Get a queue example, too.

Author: oh woops, this is a bit of copy pasta. I'll fix that

    <tr><th><code>key</code></th><td>The queue key</td></tr>
    <tr><th><code>description</code></th><td>Description of the queue</td></tr>
    <tr><th><code>url</code></th><td>Canonical API URL of the queue</td></tr>
    <tr><th><code>web_url</code></th><td>URL of the queue on Buildkite</td></tr>
    <tr><th><code>cluster_url</code></th><td>API URL of the cluster the queue belongs to</td></tr>
    <tr><th><code>dispatch_paused</code></th><td>Indicates whether the queue has paused dispatching jobs to associated agents</td></tr>
    <tr><th><code>dispatch_paused_by</code></th><td>User who paused the queue</td></tr>
    <tr><th><code>dispatch_paused_at</code></th><td>When the queue was paused</td></tr>
    <tr><th><code>dispatch_paused_note</code></th><td>The note left when the queue was paused</td></tr>
    <tr><th><code>created_at</code></th><td>When the queue was created</td></tr>
    <tr><th><code>created_by</code></th><td>User who created the queue</td></tr>
  </tbody>
</table>

### List queues

Returns a [paginated list](<%= paginated_resource_docs_url %>) of a cluster's queues.

```bash

> curl "https://api.buildkite.com/v2/organizations/{org.slug}/clusters/42f1a7da-812d-4430-93d8-1cc7c33a6bcf/queues"

Reviewer: I think for consistency with our other docs we shouldn't using real IDs in the curl examples:

```suggestion
curl "https://api.buildkite.com/v2/organizations/{org.slug}/clusters/{cluster.id}/queues"
```

  "web_url": "http://buildkite.com/organizations/test/clusters/42f1a7da-812d-4430-93d8-1cc7c33a6bcf/queues/01885682-55a7-44f5-84f3-0402fb452e66",
  "cluster_url": "http://api.buildkite.com/v2/organizations/test/clusters/42f1a7da-812d-4430-93d8-1cc7c33a6bcf",
  "dispatch_paused": true,
  "dispatch_paused_by": {
    "id": "0187dfd4-92cf-4b01-907b-1146c8525dde",
    "graphql_id": "VXNlci0tLTAxODdkZmQ0LTkyY2YtNGIwMS05MDdiLTExNDZjODUyNWRkZQ==",
    "name": "Sam Kim",
    "email": "sam@example.com",
    "avatar_url": "https://www.gravatar.com/avatar/example",
    "created_at": "2023-05-03T04:17:43.118Z"
  },
  "dispatch_paused_at": "2023-05-03T04:19:43.118Z",
  "dispatch_paused_note": "Paused for maintenance",
  "created_at": "2023-05-03T04:17:55.867Z",
  "created_by": {
    "id": "0187dfd4-92cf-4b01-907b-1146c8525dde",
    "graphql_id": "VXNlci0tLTAxODdkZmQ0LTkyY2YtNGIwMS05MDdiLTExNDZjODUyNWRkZQ==",
    "name": "Sam Kim",
    "email": "sam@example.com",
    "avatar_url": "https://www.gravatar.com/avatar/example",
    "created_at": "2023-05-03T04:17:43.118Z"
  }
}]
```

Required scope: `read_clusters`

Success response: `200 OK`

### Get a queue

```bash

> curl "https://api.buildkite.com/v2/organizations/{org.slug}/clusters/42f1a7da-812d-4430-93d8-1cc7c33a6bcf/queues/{queue.id}"

Reviewer: ```suggestion
curl "https://api.buildkite.com/v2/organizations/{org.slug}/clusters/{cluster.id}/queues/{queue.id}"
```
